syntax = "proto3";
package chatbot;

// The greeting service definition.
service ChatBot{
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc create_user (UserRequest) returns (ChatbotReply) {}
  rpc send_message (MessageRequest) returns (ChatbotReply) {}
  rpc get_message (GetRequest) returns (ChatbotReply) {}
  rpc list_users (ListRequest) returns (ChatbotReply) {}
  rpc delete_user (UserRequest) returns (ChatbotReply) {}
  rpc login_user (UserRequest) returns (ChatbotReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message UserRequest {
  string username = 1;
}

message MessageRequest {
  string username = 1;
  string message = 4;
}

message GetRequest {}

message ListRequest {
  string wildcard = 1;
}

message ChatbotReply {
  string message = 1;
}
